offset_manager.go

```diff
@@ -353,11 +349,10 @@ func (pom *partitionOffsetManager) NextOffset() (int64, string) {
 func (pom *partitionOffsetManager) AsyncClose() {
 	go func() {
 		pom.lock.Lock()
-		dirty := pom.dirty
-		pom.lock.Unlock()
+		defer pom.lock.Unlock()
 
-		if dirty {
-			<-pom.clean
+		for pom.dirty {
+			pom.clean.Wait()
```

Comment:
I haven't seen ```sync.Cond``` used before. This looks like a pretty clean solution!

This seems to solve the issue with hanging, but could ```pom.dirty``` change to true after this goroutine finishes? I suppose this would only happen if ```MarkOffset``` is called after a ```Close```, is it worth worrying about?